### ISSUE TITLE
Non-existent attribute in a blocks template -> segfault

### ISSUE THREAD

**Issue by mvartanyan** [lektor/lektor#630]
To reproduce (there may be a simpler method): 
- create a flow block 
- create a page with a flow field that can contain the flow block
- observe that everything works as expected
- change the flow block template so it contains a reference to a non-existent attribute of a method, e.g. `{{ this.url.x }}`
- observe that build crashes with a segmentation fault, taking the lektor server down along it the build is done under lektor server

Discovered on Ubuntu 18.04, reproducing on Pythons 2.7.13, 3.6.7, 3.7.1

Possibly Jinja2, not Lektor?

**Comment by nixjdm** [comment]
I'd like to confirm my understanding. In the example project, if I put `{{ this.url.x }}` anywhere in https://github.com/lektor/lektor/blob/master/example/templates/types.html, then I get a pretty useful traceback on the `/types/` page that tells me

```python traceback
...
File "/path/lektor/example/templates/types.html", line 60, in block "body"
    {{ this.url.x }}
  File "/path/lektor/lib/python3.7/site-packages/jinja2/environment.py", line 430, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'lektor.db.Page object' has no attribute 'url'
```

But if I put that line in [one of the these block files](https://github.com/lektor/lektor/tree/master/example/templates/blocks) instead, then lektor server just dies with no useful message. If that right?

**Comment by mvartanyan** [comment]
Correct, just reproduced it on the example by modifying [text_and_html.html](https://github.com/lektor/lektor/blob/master/example/templates/blocks/text_and_html.html)

**Comment by nixjdm** [comment]
Thanks for the confirmation that's what you noticed. I'll assume it's a bug over here for now. Thanks for the report!

**Comment by mvartanyan** [comment]
Some additional info. The segfault is happening [here](https://github.com/pallets/markupsafe/blob/master/src/markupsafe/_speedups.c#L338) on passing the flow block with problematic definition to `markupsafe.escape`. The hypothesis is that replacing `Py_DECREF` with `Py_XDECREF` might help with the segfault. But it won't solve the question of getting a useful traceback. Perhaps Jinja2 could catch situations when it cannot render something earlier. Go figure why it doesn't work with blocks while it works with pages. But this is as deep as I can go into this code for now, HTH.
